src/Sulu/Bundle/AdminBundle/Metadata/SchemaMetadata/ConstMetadata.php

```diff
@@ -21,16 +21,16 @@ class ConstMetadata extends PropertyMetadata
     /**
      * @param string|number|null $value
      */
-    public function __construct(string $name, bool $mandatory, $value = null)
+    public function __construct($value = null)
     {
-        parent::__construct($name, $mandatory);
+        parent::__construct();
+
         $this->value = $value;
     }
 
-    public function toJsonSchema(): ?array
+    public function toJsonSchema(): array
     {
         return [
-            'name' => $this->getName(),
```

Comment:
In the `ArrayMetadata` you've called `parent::toJsonSchema` as well. Any reason this is not happening in this case?

Reply:
The SchemaMetadata returns `['required' => []]`, if it's real schema is empty, because it always needs to return an associative array, otherwise the admin ui would treat it as array instead of an object and would break.

The `ArrayMetadata` initializes `SchemaMetadata` with type `array`, so `parent::toJsonSchema()` will return ['type' => 'array'].

But the `ConstMetadata` initializes `SchemaMetadata` with type `null`, so `parent::toJsonSchema()` would return ['required' => []] and we don't want the final json schema to look like `['required' => [], 'const' => 'any-value']`

Reply:
Ok, that's a bit implicit, but I get it... Let's leave it this way.